api/include/opentelemetry/version.h

```diff
@@ -6,6 +6,13 @@
 #include "opentelemetry/common/macros.h"
 #include "opentelemetry/detail/preprocessor.h"
 
+#if defined(OPENTELEMETRY_STL_VERSION)
+#  if OPENTELEMETRY_STL_VERSION > (__cplusplus/100)
+#    pragma message OPENTELEMETRY_STRINGIFY(OPENTELEMETRY_STL_VERSION) " vs. " OPENTELEMETRY_STRINGIFY(__cplusplus)
+#    error "OPENTELEMETRY_STL_VERSION set to version newer than compilation version."
```

Comment:
```suggestion
#    error "OPENTELEMETRY_STL_VERSION is set to a version newer than the compiler version."
```

Reply:
"the compiler version" is ambiguous. Is it saying the version of GCC/Clang it self? The version of C++ _supported_ by the compiler? The version currently being _used_ by the compiler? I'm not sure that "compilation version" is any more accurate, but it at least doesn't invite that first incorrect interpretation. Any suggestions on how to avoid that?

I'll go with whatever you want, but if we can't come up with something clear, then I'd rather something that makes people think a bit before they make a **bad** assumption.

Reply:
We can use `_MSVC_LANG` to check C++ version without `/Zc:__cplusplus` using MSVC.